internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -46,12 +54,15 @@ func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 	switch {
 	case bytes.Equal(sslMode, []byte("disable")):
 		connDetails.SSLMode = string(credentials["sslmode"])
+		break
 	case bytes.Equal(sslMode, []byte("enable")):
+		connDetails.parseSSLOptions(credentials)
```

Comment:
I think we can refactor this so that the `parseSSLOptions` is separated out and the switch can be concerned wholly with setting SSLMode. This can be done by moving the invocation of `parseSSLOptions` to after the switch:

```go
if connDetails.SSLMode != "disable" {
	connDetails.parseSSLOptions(credentials)
}
```

Reply:
re: the `enable` case, it isn't a valid sslmode, I think that should be `require`.